[Joshua Liang XingYa] iP

joshualiangxy opened the pull request.

Commit:
Add Gradle support

joshualiangxy committed:
Initial commit

joshualiangxy committed:
Add list command, add task command and Task class

joshualiangxy committed:
Add done command

joshualiangxy committed:
Add Task subclasses and update addTask method for the new classes

joshualiangxy committed:
Update runtime.bat and runtime.sh to compile all .java files

joshualiangxy committed:
Add exception classes

joshualiangxy committed:
Add deleteTask method and InvalidTaskException class

joshualiangxy committed:
Add tests and update exception classes to be more suitable

joshualiangxy committed:
Remove logo

joshualiangxy committed:
Add write methods to save tasks as csv

joshualiangxy committed:
Add setting of tasks from data file

joshualiangxy committed:
Update .gitignore to ignore data file

joshualiangxy committed:
Update .gitignore file to ignore data file

joshualiangxy committed:
Add proper support for datetime and update tests

joshualiangxy committed:
Merge branch 'branch-Level-7'

joshualiangxy committed:
Merge branch 'branch-Level-8'

joshualiangxy committed:
Fix merge conflicts regarding writing and saving datetime

joshualiangxy committed:
Fix read and write methods

joshualiangxy committed:
Refactor code

Add Ui, TaskList, Parser and Storage classes as well as Command
interfaces for abstraction

joshualiangxy committed:
Refactor project files into packages

joshualiangxy committed:
Configure junit and test Parser

joshualiangxy committed:
Configure jar file

joshualiangxy committed:
Add documentation for the application

joshualiangxy committed:
Update styling to fit coding standards

joshualiangxy committed:
Add command to find tasks in list

joshualiangxy committed:
Merge branch 'branch-A-JavaDoc'

joshualiangxy committed:
Merge branch 'branch-A-CodingStandard'

joshualiangxy committed:
Merge branch 'branch-Level-9'

joshualiangxy committed:
Merge branch 'add-gradle-support' into master

joshualiangxy committed:
Configure gradle build for junit tests

joshualiangxy committed:
Configure gradle checkstyle plugin

joshualiangxy committed:
Merge branch 'branch-A-CheckStyle' into master

joshualiangxy committed:
Fix styling according to checkstyle

joshualiangxy committed:
Update ParserTest with find command

joshualiangxy committed:
Set up javafx

joshualiangxy committed:
Set up layout for gui

joshualiangxy committed:
Make gui echo user

joshualiangxy committed:
Move Duke to gui package and add DialogBox class

joshualiangxy committed:
Add dialog box to differentiate user and duke

joshualiangxy committed:
Link gui to the main program logic

joshualiangxy committed:
Merge branch 'branch-Level-10' into master

joshualiangxy committed:
Add spacing in gui greeting and errors

duynguyen24501 commented on src/main/command/AddDeadlineCommand.java:
```diff
@@ -0,0 +1,59 @@
+     * @return true.
+     */
+    @Override
+    public boolean hasCommand() {
```
Will it be clearer if we rename it to something like isValidCommand()? 🤔

duynguyen24501 commented on src/main/command/AddTodoCommand.java:
```diff
@@ -0,0 +1,56 @@
+ * @version v0.1
+ * @since v0.1
+ */
+public class AddTodoCommand implements Command {
```
Will it be better if we change class name "AddTodoCommand" which sounds like a verb to TodoCommand?

duynguyen24501 commented on src/main/parser/Parser.java:
```diff
@@ -0,0 +1,158 @@
+        }
+    }
+
+    private static Command parseAdd(String[] input)
```
You can consider naming it as "parseCommand" instead of "parseAdd" to make it a bit clearer? 🤔

joshualiangxy replied:
The `parseAdd` method is meant for the parsing only commands to add new tasks, so `parseCommand` would be too broad of a name for this method. Admittedly, `parseAdd` isn't a very good name either, so I would be happy if you had more suggestions!

duynguyen24501 commented on src/main/task/Event.java:
```diff
@@ -0,0 +1,64 @@
+                time.format(FORMATTER));
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj instanceof Event) {
+            Event o = (Event) obj;
+            return super.equals(o) && this.time.equals(o.time);
+        }
+        return false;
+    }
```
The idea of overriding equals method is great! 👍

duynguyen24501 commented on src/main/ui/Ui.java:
```diff
@@ -0,0 +1,113 @@
+     * Returns the string bidding the user farewell.
+     * @return the string bidding the user farewell.
+     */
+    public String printExit() {
```
Just a small thing, I think we should change it to "printExitMessage"?

duynguyen24501 reviewed (commented):
Comment on code quality(Naming) 😃 Overall most of your code follows coding quality already. Good job! 👍

junhui-phoon commented on src/main/command/FindCommand.java:
```diff
@@ -0,0 +1,63 @@
+     */
+    @Override
+    public String execute(Ui ui, TaskList tasks) {
+        TaskList found = new TaskList();
```
Perhaps you could change 'found' to 'foundTasks' etc? Plural form should be used to represent a collection of objects :)

junhui-phoon commented on src/main/storage/Storage.java:
```diff
@@ -0,0 +1,100 @@
+            String taskType = task[0];
+            String taskTime = task[1];
+            String taskName = task[3];
+            boolean taskDoneState = task[2].equals("1");
```
Maybe you could rename 'taskDoneState' to 'isTaskDone'?

junhui-phoon commented:
Hello! Great job on adhering to the coding standard 👍🏻

joshualiangxy committed:
Improve coding standards according to reviews

joshualiangxy committed:
Refactor Duke to Stuff and add tests for Ui class

joshualiangxy committed:
Add unit tests for Deadline class

joshualiangxy committed:
Add unit tests for the Event class

joshualiangxy committed:
Add unit tests for Todo class

joshualiangxy committed:
Add unit tests for Task class

joshualiangxy committed:
Add unit tasks for TaskList class

joshualiangxy committed:
Add unit tests for DeadlineCommand

joshualiangxy committed:
Add unit tests for DeleteCommand class

joshualiangxy committed:
Add unit tests for DoneCommand class

joshualiangxy committed:
Add unit tests for EventCommand class

joshualiangxy committed:
Add unit tests for ExitCommand class

joshualiangxy committed:
Add unit tests for FindCommand class

joshualiangxy committed:
Add unit tests for ListCommand class

joshualiangxy committed:
Add unit tests for TodoCommand class

joshualiangxy committed:
Add unit tests for StuffException class

joshualiangxy committed:
Add unit tests for EmptyMessageException class

joshualiangxy committed:
Add unit tests for InvalidDateException class

joshualiangxy committed:
Add unit tests for InvalidDeadlineFormatException

joshualiangxy committed:
Add unit tests for InvalidEventFormatException

joshualiangxy committed:
Add unit tests for InvalidTaskException class

joshualiangxy committed:
Add unit tests for UnknownCommandException class

joshualiangxy committed:
Reconfigure build.gradle to use Launcher

This fixes the error where javafx is not packaged into the jar

joshualiangxy committed:
Update javadoc for new release

joshualiangxy committed:
Add assertions

joshualiangxy committed:
Improve code quality and improve find command

Make the find command work with multiple search terms

joshualiangxy committed:
Update javadocs

joshualiangxy committed:
Add artificial load time

joshualiangxy committed:
Implement streams for printing task lists

The list and find commands is able to make use of streams to print out
the task lists

joshualiangxy committed:
Update javadoc

joshualiangxy committed:
Merge pull request #1 from joshualiangxy/branch-A-Assertions

Add assertions

joshualiangxy committed:
Fix naming for greeting and error tests

joshualiangxy committed:
Merge branch 'master' into branch-A-CodeQuality

joshualiangxy committed:
Merge pull request #2 from joshualiangxy/branch-A-CodeQuality

Improve code quality and find command

joshualiangxy committed:
Merge branch 'master' into branch-A-Lambdas

joshualiangxy committed:
Merge pull request #3 from joshualiangxy/branch-A-Lambdas

Add artificial load time

joshualiangxy committed:
Merge branch 'master' into branch-A-Streams

joshualiangxy committed:
Merge pull request #4 from joshualiangxy/branch-A-Streams

Implement streams for printing task lists

joshualiangxy committed:
Add workflow for gradle

joshualiangxy committed:
Remove text-ui-test from gradle.yml file

joshualiangxy committed:
Remove string literals in Ui class

joshualiangxy committed:
Add support for recurring tasks

joshualiangxy committed:
Fix tasks not being set with proper done state

joshualiangxy committed:
Add unit tests for recurring deadline and events

joshualiangxy committed:
Update javadocs

joshualiangxy committed:
Merge branch 'branch-B-RecurringTasks' into master

joshualiangxy committed:
Add tests for parsing commands with options

joshualiangxy committed:
Add support for tagging and searching for tasks

joshualiangxy committed:
Add unit tests for tagging functionality

joshualiangxy committed:
Merge branch 'branch-C-Tagging' into master

joshualiangxy committed:
Add icon to gui

joshualiangxy committed:
Refactor padding from Stuff to DialogBox class

joshualiangxy committed:
Improve false load time in replies

False load times are now randomly generated from 250 to 750ms.

joshualiangxy committed:
Remove unused loading ImageView

joshualiangxy committed:
Update docs/README.md with a user guide
